executor/load_data.go

```diff
@@ -217,76 +216,40 @@ func processStream(ctx context.Context, readerFn func() ([]byte, error), loadDat
 				zap.Stack("stack"))
 		}
 		if err != nil || r != nil {
-			loadDataInfo.ForceQuit()
+			loadDataInfo.forceQuit()
 		} else {
 			loadDataInfo.CloseTaskQueue()
 		}
 		wg.Done()
 	}()
-	for {
-		curData, err = readerFn()
-		if err != nil {
-			if terror.ErrorNotEqual(err, io.EOF) {
-				logutil.Logger(ctx).Error("read data for LOAD DATA failed", zap.Error(err))
-				break
-			}
-			err = nil
-		}
-		if len(curData) == 0 {
-			loadDataInfo.Drained = true
-			shouldBreak = true
-			if len(prevData) == 0 {
-				break
-			}
-		}
-		select {
-		case <-loadDataInfo.QuitCh:
-			err = errors.New("processStream forced to quit")
-		default:
-		}
-		if err != nil {
-			break
-		}
-		// prepare batch and enqueue task
-		prevData, err = insertDataWithCommit(ctx, prevData, curData, loadDataInfo)
-		if err != nil {
-			break
-		}
-		if shouldBreak {
-			break
-		}
-	}
-	if err != nil {
-		logutil.Logger(ctx).Error("load data process stream error", zap.Error(err))
-		return
-	}
-	if err = loadDataInfo.EnqOneTask(ctx); err != nil {
-		logutil.Logger(ctx).Error("load data process stream error", zap.Error(err))
-		return
-	}
-}
 
-func insertDataWithCommit(ctx context.Context, prevData,
-	curData []byte, loadDataInfo *LoadDataInfo) ([]byte, error) {
-	var err error
-	var reachLimit bool
+	// TODO: use parser interface
+	csvParser, err = mydump.NewCSVParser(
+		ctx,
+		loadDataInfo.GenerateCSVConfig(),
+		reader,
+		int64(config.ReadBlockSize),
+		nil,
+		false,
+		// TODO: support charset conversion
+		nil)
+	csvParser.SetLogger(log.Logger{Logger: logutil.Logger(ctx)})
+
 	for {
-		prevData, reachLimit, err = loadDataInfo.InsertData(ctx, prevData, curData)
+		// prepare batch and enqueue task
+		err = loadDataInfo.ReadRows(ctx, csvParser)
 		if err != nil {
-			return nil, err
+			logutil.Logger(ctx).Error("load data process stream error", zap.Error(err))
+			return
 		}
-		if !reachLimit {
-			break
+		if loadDataInfo.curBatchCnt == 0 {
+			return
 		}
-		// push into commit task queue
-		err = loadDataInfo.EnqOneTask(ctx)
-		if err != nil {
-			return prevData, err
+		if err = loadDataInfo.enqOneTask(ctx); err != nil {
+			logutil.Logger(ctx).Error("load data process stream error", zap.Error(err))
```

Comment:
maybe make the error message more specific to the function (i.e., `enqOneTask`) to make it easier to locate the error position according to the log.